src/Adapters/Storage/Mongo/MongoCollection.js

```diff
@@ -112,7 +112,20 @@ export default class MongoCollection {
       findOperation = findOperation.project(keys);
     }
 
-    if (caseInsensitive) {
+    if (query.username?.$regex || query.email?.$regex) {
+      const indexName = query.username?.$regex ? 'username_1' : 'email_1';
+      findOperation = await new Promise((resolve, reject) => {
+        this._mongoCollection.indexExists(indexName, (error, result) => {
+          if (error) {
+            reject(error);
+          } else {
+            result
+              ? resolve(findOperation.hint(indexName))
+              : resolve(findOperation.hint({ $natural: 1 }));
```

Comment:
This is how you tell a Mongo query to not use an index at all.

Reply:
Hmm fair points. I guess we should just assume that if the "better" index doesn't exist, then we the user has done something custom to their index setup and we shouldn't assume anything.

Reply:
Well, Parse does support the use of our own hinting.  I think as long as that overrides whatever we set as the default we are ok.  So we choose no index by default because that will definitely work better than the 2 we know about and let the user specify something else if they really do have a custom index that makes sense to use.

I don't think there is any chance the 2 indexes don't exist.  Whenever the server starts up it creates both of them.  I suppose someone could have a script to delete them after startup, but that is NOT recommended and we shouldn't support that hack, imho.

Reply:
Ok I've updated the code to do the following steps:
1. Check whether we are doing a regex query on username or email
2. If so, check whether we have a `case_insensitive` index on that queried field
3. If so, check whether we have a normal non-collated index (named `{field}_1`) that we can use instead of the case_insensitive index
4. If so, force use of that non-collated index with hint.

If at any point any of the above checks are not true, we continue like normal allowing the user to pass a hint if they want.